### PR TITLE
Fix broken CheckStyle build and static analysis violations.

### PR DESCRIPTION
**Pull request by motlin**

### CODE REVIEW

File: checkstyle-configuration.xml [hunk 1]
```diff
@@ -81,7 +85,15 @@
...
-        <module name="OperatorWrap" />
+        <module name="OperatorWrap">
+            <property name="tokens" value="ASSIGN, DIV_ASSIGN, PLUS_ASSIGN, MINUS_ASSIGN, STAR_ASSIGN, MOD_ASSIGN, SR_ASSIGN, BSR_ASSIGN, SL_ASSIGN, BXOR_ASSIGN, BOR_ASSIGN, BAND_ASSIGN" />
+            <property name="option" value="eol" />
+        </module>
+
+        <module name="OperatorWrap">
+            <property name="tokens" value="QUESTION, COLON, EQUAL, NOT_EQUAL, DIV, PLUS, MINUS, STAR, MOD, SR, BSR, GE, GT, SL, LE, LT, BXOR, BOR, LOR, BAND, LAND, LITERAL_INSTANCEOF, TYPE_EXTENSION_AND, METHOD_REF" />
+            <property name="option" value="nl" />
+        </module>
```
Comment (motlin) [inline checkstyle-configuration.xml]: This check must have been broken in older versions of CheckStyle.

File: checkstyle-configuration.xml [hunk 2]
```diff
@@ -109,13 +122,17 @@
...
         <!-- Modifier Checks                                    -->
         <!-- See http://checkstyle.sf.net/config_modifiers.html -->
         <module name="ModifierOrder" />
         <module name="RedundantModifier" />
 
         <module name="LeftCurly">
-            <property name="option" value="nl" />
+            <property name="option" value="nl"/>
+            <!--TODO: Remove this, to include the LAMBDA token-->
+            <property name="tokens" value="INTERFACE_DEF, CLASS_DEF, ANNOTATION_DEF, ENUM_DEF, CTOR_DEF, METHOD_DEF, ENUM_CONSTANT_DEF, LITERAL_WHILE, LITERAL_TRY, LITERAL_CATCH, LITERAL_FINALLY, LITERAL_SYNCHRONIZED, LITERAL_SWITCH, LITERAL_DO, LITERAL_IF, LITERAL_ELSE, LITERAL_FOR, STATIC_INIT, OBJBLOCK"/>
```
Comment (motlin) [inline checkstyle-configuration.xml]: Older versions of CheckStyle didn't enforce any convention for lambda curly braces, and so our codebase has a lot of inconsistencies. If we choose to fix them, I'd want to do so in a separate commit. For now, I listed all tokens as eligible for checking except for lambdas.

File: checkstyle-configuration.xml [hunk 3]
```diff
@@ -186,8 +204,8 @@
                 value="System.out.println." />
         </module>
 
-        <module name="Regexp">
-            <property name="illegalPattern" value="true" />
+        <module name="RegexpSinglelineJava">
+            <property name="ignoreComments" value="true" />
```
Comment (motlin) [inline checkstyle-configuration.xml]: Finally, if you add System.out.println and then comment it out, the build will pass.

File: jcstress-tests/pom.xml [hunk 4]
```diff
@@ -60,8 +60,8 @@
                             <mainClass>org.openjdk.jcstress.Main</mainClass>
                         </transformer>
                         <transformer implementation="org.apache.maven.plugins.shade.resource.AppendingTransformer">
-			    <resource>META-INF/TestList</resource>
-			 </transformer>
+                            <resource>META-INF/TestList</resource>
+                        </transformer>
```
Comment (motlin) [inline jcstress-tests/pom.xml]: tabs to spaces

File: p2-feature/org.eclipse.collections.feature/pom.xml [hunk 5]
```diff
@@ -11,47 +11,47 @@
...
+                        <phase>package</phase>
+                        <goals>
+                            <goal>p2-metadata</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
 </project>
```
Comment (motlin) [inline p2-feature/org.eclipse.collections.feature/pom.xml]: tabs to spaces

File: pom.xml [hunk 6]
```diff
@@ -420,7 +420,7 @@
                     <dependency>
                         <groupId>com.puppycrawl.tools</groupId>
                         <artifactId>checkstyle</artifactId>
-                        <version>6.10.1</version>
+                        <version>${checkstyle.version}</version>
```
Comment (motlin) [inline pom.xml]: This is the critical fix. This was preventing many checks from running.

Comment (nikhilnanivadekar) [reply pom.xml]: Good catch! I wonder how this was missed for so long! 👍

File: eclipse-collections/src/main/java/org/eclipse/collections/impl/map/mutable/ConcurrentHashMapUnsafe.java [hunk 7]
```diff
@@ -242,9 +242,9 @@ private int hash(Object key)
         int h = key.hashCode();
         h ^= (h >>> 18) ^ (h >>> 12);
         return h ^ (h >>> 10);
-//        h ^= h >>> 20 ^ h >>> 12;
-//        h ^= h >>> 7 ^ h >>> 4;
-//        return h;
+        // h ^= h >>> 20 ^ h >>> 12;
```
[the remark this answers is not in the document]
Comment (motlin) [reply eclipse-collections/src/main/java/org/eclipse/collections/impl/map/mutable/ConcurrentHashMapUnsafe.java]: I don't know, I just fixed checkstyle errors.
